src/Simplify.elm

```diff
@@ -5156,62 +5163,59 @@ listIntersperseCompositionChecks checkInfo =
-                            Just
-                                (returnsArgError (qualifiedToString checkInfo.fn ++ " with [] as the second argument")
-                                    { arg = listArgToTheLeft
-                                    , argRepresents = "first list argument"
-                                    }
-                                    checkInfo
-                                )
+                Just secondArg_ ->
+                    if collection.empty.is checkInfo.lookupTable secondArg_ then
+                        Just
+                            (Rule.errorWithFix
+                                { message = "Unnecessary append with " ++ descriptionForIndefinite collection.empty.description
+                                , details = [ "You can replace this call by the " ++ collection.represents ++ " itself." ]
+                                }
+                                checkInfo.fnRange
+                                (keepOnlyFix { parentRange = checkInfo.parentRange, keep = Node.range checkInfo.firstArg })
+                            )
 
-                        Just (_ :: _) ->
-                            case AstHelpers.getListLiteral listArgToTheLeft of
-                                Just (_ :: _) ->
-                                    Just
-                                        (Rule.errorWithFix
-                                            { message = "Appending literal lists could be simplified to be a single List"
-                                            , details = [ "Try moving all the elements into a single list." ]
-                                            }
-                                            checkInfo.fnRange
-                                            [ Fix.removeRange { start = (Node.range listArgToTheRight).end, end = checkInfo.parentRange.end }
-                                            , Fix.replaceRangeBy
-                                                { start = checkInfo.parentRange.start, end = startWithoutBoundary (Node.range listArgToTheRight) }
-                                                ("[" ++ checkInfo.extractSourceCode (rangeWithoutBoundaries (Node.range listArgToTheLeft)) ++ ",")
-                                            ]
-                                        )
+                    else
+                        case collection.fromListLiteralRange checkInfo.lookupTable secondArg_ of
+                            Just literalListRangeSecond ->
+                                case collection.fromListLiteralRange checkInfo.lookupTable checkInfo.firstArg of
+                                    Just literalListRangeFirst ->
+                                        Just
+                                            (Rule.errorWithFix
+                                                { message = "Appending literal " ++ collection.represents ++ "s could be simplified to be a single " ++ collection.represents
```

Comment:
As above, I'd prefer using `... checkInfo.fn ++ " on ..."` instead of `"Appending ..."` to be more precise and situational.
Also not sure I'd call `Array.fromList [ 1, 2, 3 ]` a "literal".

Reply:
Yeah, was going to do this in the next PR.
I agree the term "literal" is not perfect, but I think it's what I think comes closest. Let's use this until we come up with something better.

Reply:
I would not just use `Tuple.second` (which btw should probably be `AstHelpers.qualifiedName`) as just the name here. Why not use the qualified name?

Reply:
It was to keep the same error message as for the union checks and the list append checks. But yeah, let's change it :+1: